src/lib/chat/chat-message.ts

```diff
@@ -0,0 +1,125 @@
+import { Message } from '../../store/messages';
+import { ChatMember } from './types';
+
+const DEFAULT_MEDIA_TYPE = 'image';
+
+export const getChatMember = (sbMember: any): ChatMember => {
+  let chatMember: ChatMember = {
+    userId: sbMember.userId,
+    firstName: sbMember.nickname,
+    lastName: '',
+    profileImage: sbMember.profileUrl,
+    profileId: null,
+  };
+  // Some requests don't return the lastSeenAt value. If it's not valid
+  // then don't set it here so that it doesn't override any existing state
+  // later on
+  if (sbMember.lastSeenAt && sbMember.lastSeenAt > 0) {
+    chatMember.lastSeenAt = new Date(sbMember.lastSeenAt);
+  }
+
+  if (sbMember.connectionStatus === 'online') {
+    chatMember.isOnline = true;
+  }
+  if (sbMember.connectionStatus === 'offline') {
+    chatMember.isOnline = false;
+  }
+
+  try {
+    const { profileImage, profileId, firstName, lastName } = JSON.parse(sbMember.profileUrl);
+    chatMember = {
+      ...chatMember,
+      firstName,
+      lastName,
+      profileImage,
+      profileId,
+    };
+  } catch (err) {}
+
+  return chatMember;
+};
+
+function getSender(sendbirdUser, isAdmin) {
+  if (sendbirdUser) {
+    return getChatMember(sendbirdUser);
+  } else if (isAdmin) {
```

Comment:
Wast this `isAdmin` condition carried over from ZERO Web? I think we can remove it from zOS.